compiler/src/iree/compiler/Codegen/Common/StripCompilationInfoPass.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/Passes.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-codegen-strip-compilation-info"
+
+namespace mlir::iree_compiler {
+
+#define GEN_PASS_DEF_STRIPCOMPILATIONINFOPASS
+#include "iree/compiler/Codegen/Common/Passes.h.inc"
+
+namespace {
+
+// Checks whether the funcOp has any compilation info.
+LogicalResult hasCompilationInfo(func::FuncOp funcOp) {
+  if (getTranslationInfo(funcOp))
+    return success();
```

Comment:
This assumption is not correct. We may have lowering config / compilation info but not translation info. This would be true for modules configured by TD scripts.

Reply:
For the case you proposed,  this condition cannot pass and then the function will continue to check whether it has lowering config/translation info.